src/view/actions/sendEvent.jsx

```diff
@@ -26,22 +26,78 @@ import getInstanceOptions from "../utils/getInstanceOptions";
 import singleDataElementRegex from "../constants/singleDataElementRegex";
 import "./sendEvent.styl";
 import InfoTipLayout from "../components/infoTipLayout";
+import DecisionScopesComponent from "../components/decisionScopesComponent";
 
+const decisionScopesOptions = {
+  CONSTANT: "constant",
+  DATA_ELEMENT: "dataElement"
+};
+
+const filterDecisionScopes = scopes => {
+  return scopes.filter(s => s !== "");
+};
+
+const getDecisionScopes = values => {
+  if (
+    values.option === decisionScopesOptions.DATA_ELEMENT &&
+    values.decisionScopesDataElement
+  ) {
+    return values.decisionScopesDataElement;
+  }
+
+  if (
+    values.option === decisionScopesOptions.CONSTANT &&
+    values.decisionScopesArray.length > 0
+  ) {
+    const scopes = filterDecisionScopes(values.decisionScopesArray);
+    if (scopes.length > 0) {
+      return scopes;
+    }
+  }
+  return undefined;
+};
+
+const getInitialDecisionScopesData = decisionScopes => {
+  if (Array.isArray(decisionScopes)) {
+    return {
+      option: decisionScopesOptions.CONSTANT,
+      decisionScopesArray: decisionScopes,
+      decisionScopesDataElement: ""
+    };
+  }
+  if (typeof decisionScopes === "string") {
+    return {
+      option: decisionScopesOptions.DATA_ELEMENT,
+      decisionScopesDataElement: decisionScopes,
+      decisionScopesArray: [""]
+    };
+  }
+  return {
+    option: decisionScopesOptions.DATA_ELEMENT,
```

Comment:
I think the default should be a CONSTANT, but I'm curious to see what others think.

Reply:
That's what I would have guessed, but wasn't sure.

Reply:
ok, will change that, thank you